frontend/javascripts/oxalis/view/right-border-tabs/tree_hierarchy_view.tsx

```diff
@@ -157,8 +151,15 @@ function TreeHierarchyView(props: Props) {
       ).map((node) => node.key);
       expandedKeys = _.without(expandedKeys, ...subGroupKeys);
     }
-
-    setExpandedNodeKeys(expandedKeys);
+    const expandedKeySet = new Set(expandedKeys);
```

Comment:
in line 152, there is a _.without which is a bit expensive since it iterates over the entire array. since you convert the array into a Set anyway, you can do the removal on the set and afterwards convert to an array if needed.